proposals/accepted/047-standard-library-reform.rst

```diff
@@ -0,0 +1,559 @@
+We will thus need more than one standard library.
+
+Platform-specific functionality should be exposed in ways that make sense in Haskell, not C.
+Traditional libc idioms and "lowest common denominator" practice should be skipped when it does not make sense in a Haskell context.
+It should be possible to use WASM and WASI without any "libc".
+
+**Problem 4**: Breaking changes have to coupled, not staggered, with GHC versions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Wishful thinking would have it that we can just *stop* doing breaking changes, forever.
+But requirements change, and mistakes are made.
+Issues will arise in the standard library and we will wish to fix them, because whatever the cost is to existing programs (which we can still attempt to mitigate) is outweighed by the benefit to future programs.
+
+However, if the standard library version is tied to GHC version, we have no choice but to do the breaking change coupled with a compiler version.
+Gabriella Gonzalez laid out the case in `Release early and often <https://www.haskellforall.com/2019/05/release-early-and-often.html>`_ on why coupling changes, especially breaking changes, together is bad, and I will cite that rather than restate the argument.
+For those reasons we shouldn't do that here with the standard library and GHC.
+
+Solution criteria
+^^^^^^^^^^^^^^^^^
+
+Changes in the standard library in the compiler should always be staggered.
+It should be possible to upgrade the compiler with only a minor version change or less in the standard library.
+It should likewise be possible to upgrade a major version change in the standard library without breaking a compiler.
+
+**Problem 5**: Popular and uncontroversial machinery like ``Text`` not available from the standard library
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There has been much grumbling over the years that popular items like ``Text`` are not in the standard library.
+Items like these are expected to be in languages' standard libraries and elsewhere indeed are found there.
+
+Now, it is one thing for a standard library to be minimal, and say not offer any string type or operations on that.
+That would not be so bad.
+What is worse is that ``base`` does offer ``String``, and furthermore operations on ``String``.
+The problem is thus not so much that it is inconvenient to grab the ``Text``-based functionality from elsewhere, as it is that ``base`` has a foot-gun in offering alternatives that should be *avoided*.
+Standard libraries which *mislead* the user as to what they ought to do are worse than standard libraries which stay mum altogether.
+
+Solution criteria
+^^^^^^^^^^^^^^^^^
+
+Firstly, do not offer bad alternatives in the standard library that users should not use.
```

Comment:
Deleting `String` sounds like a great way to break _every Haskell program on the planet_. Do you really mean that?